url27a7ae3508_serkan_ozal_ocean_of_memories_tgz-pJ8-com_zeroturnaround_rebellabs_oceanofmemories_article1_objectlayout_ObjectMemoryLayoutDemoJ8/src/com/zeroturnaround/rebellabs/oceanofmemories/common/util/ReflectionUtil.java

```diff
@@ -0,0 +1,349 @@
+            } catch (NoSuchFieldException e) {
+                logger.debug("Unable to get field " + fieldName
+                        + " from class " + cls.getName() + ". "
+                        + "If there is any super class for class "
+                        + cls.getName()
+                        + ", it will be tried for getting field ...");
+            }
+            if (field != null) {
+                field.setAccessible(true);
+                return field;
+            }
+            cls = cls.getSuperclass();
+        }
+        return null;
+    }
+
+    @SuppressWarnings("unchecked")
+    public static List<Field> getAllFields(Class<?> cls) {
+        List<Field> fields = new ArrayList<Field>();
+        createFields(cls, fields);
+        return fields;
+    }
+
+    @SuppressWarnings("unchecked")
+    public static List<Field> getAllFields(Class<?> cls,
+            Class<? extends Annotation>... annotationFilters) {
+        List<Field> fields = new ArrayList<Field>();
+        createFields(cls, fields, annotationFilters);
+        return fields;
+    }
+
+    @SuppressWarnings("unchecked")
+    private static void createFields(Class<?> cls, List<Field> fields,
+            Class<? extends Annotation>... annotationFilters) {
+        if (cls == null || cls.equals(Object.class)) {
+            return;
+        }
+
+        Class<?> superCls = cls.getSuperclass();
+        createFields(superCls, fields, annotationFilters);
+
+        for (Field f : cls.getDeclaredFields()) {
+            f.setAccessible(true);
+            if (annotationFilters == null || annotationFilters.length == 0) {
+                fields.add(f);
+            } else {
+                for (Class<? extends Annotation> annotationFilter : annotationFilters) {
+                    if (f.getAnnotation(annotationFilter) != null) {
+                        fields.add(f);
+                        break;
+                    }
+                }
+            }
+        }
+    }
+
+    public static Method getMethod(Class<?> cls, String methodName) {
+        while (cls != null && cls.equals(Object.class) == false) {
+            Method method = null;
+            for (Method m : cls.getDeclaredMethods()) {
+                if (m.getName().equals(methodName)) {
+                    method = m;
+                    break;
+                }
+            }
+            if (method == null) {
+                logger.warn("Unable to get method " + methodName
+                        + " from class " + cls.getName() + ". "
+                        + "If there is any super class for class "
+                        + cls.getName()
+                        + ", it will be tried for getting method ...");
+            } else {
+                method.setAccessible(true);
+                return method;
+            }
+            cls = cls.getSuperclass();
+        }
+        return null;
+    }
+
+    public static Method getMethod(Class<?> cls, String methodName,
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 12. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.